spec/lib/vcr/library_hooks/typhoeus_spec.rb

```diff
@@ -110,46 +110,86 @@ def make_request
 
-    it 'records and replays correctly' do
-      expect(on_headers_counter).to receive(:increment).exactly(2).times
-      expect(on_body_counter).to receive(:increment).exactly(2).times
+    it { expect(request.tap { |r| r.on_headers {} }).not_to be_streaming }
 
-      recorded = make_request
-      played_back = make_request
+    it { expect { |b| on_headers(&b) }.to yield_with_args(have_attributes(headers: include('Content-Length' => '18'))) }
+    it { expect { |b| on_headers(&b) }.to yield_with_args(have_attributes(headers: match_array(on_headers.headers))) }
 
-      expect(recorded.body).to eq('Localhost response')
-      expect(played_back.body).to eq(recorded.body)
-    end
+    it { expect(on_headers).to have_attributes(headers: include("Content-Length" => "18")) }
+    it { expect(on_headers).to have_attributes(headers: match_array(on_headers.headers))  }
   end
 
-  context 'when using on_body callback returning :abort' do
-    def make_request
-      VCR.use_cassette('no_body') do
-        request = Typhoeus::Request.new("http://localhost:#{VCR::SinatraApp.port}/localhost_test")
-        request.on_body { next :abort }
-        request.run
-      end
+  context 'when using on_body callback' do
+    def on_body(&callback)
+      callback = Proc.new {} unless block_given?
+
+      VCR.use_cassette('no_body') { request.tap { |r| r.on_body(&callback) }.run }
     end
 
-    it 'records and replays correctly' do
-      recorded = make_request
-      played_back = make_request
+    def request
+      Typhoeus::Request.new("http://localhost:#{VCR::SinatraApp.port}/localhost_test")
+    end
 
-      expect(recorded.body).to eq('Localhost response')
-      expect(played_back.body).to eq(recorded.body)
+    it { expect(request.tap { |r| r.on_body {} }).to be_streaming }
+
+    it { expect(on_body).to have_attributes(body: 'Localhost response') }
+    it { expect(on_body).to have_attributes(body: on_body.body)  }
+
+    it { expect { |b| on_body(&b) }.to yield_with_args('Localhost response', have_attributes(body: '')) }
+    it { expect { |b| on_body(&b) }.to yield_with_args(on_body.body, have_attributes(body: 'Localhost response')) }
+
+    it { expect(on_body { next :abort }).to have_attributes(body: on_body.body)  }
+    it { expect(on_body { next :abort }).to have_attributes(body: on_body { next :abort }.body)  }
+    it { expect(on_body).to have_attributes(body: on_body { next :abort }.body)  }
+
+    context 'when re-using single request instance' do
+      let(:request) { Typhoeus::Request.new("http://localhost:#{VCR::SinatraApp.port}/localhost_test") }
+
+      it { expect(on_body).to have_attributes(body: 'Localhost response') }
+      it { expect { |b| on_body(&b) }.to yield_with_args('Localhost response', have_attributes(body: '')) }
+
+      context 'after recording cassette (first request)' do
+        before { on_body }
+
+        it { expect(on_body).to have_attributes(body: 'Localhost responseLocalhost response') } # FIXME
```

Comment:
Curious about these: is this change exposing the re-recording issue, or causing it? Should the FIXME be repaired in this PR?

Reply:
My intention was to expose what to me looks like an issue.
I'd prefer to address it separately please, if that's alright of course.